R/caretList.R

```diff
@@ -322,3 +341,16 @@ predict.caretList <- function(object, newdata = NULL, ..., verbose = FALSE){
 
   return(preds)
 }
+
+#' @title Index a caretList
+#' @description Index a caret list to extract caret models into a new caretList object
+#"
+#' @param object an object of class caretList
+#' @export
+`[.caretList` <- function(object, index){
+  newObject <- object
+  class(newObject) <- "list"
+  newObject <- newObject[index]
+  class(newObject) <- "caretList"
+  return(newObject)
```

Comment:
I found a better approach, i'm add it to the pull request. Inherit the list class in the caretList constructor. Then there is no need to even define `[` operators and you get all the list methods.